src/lib/timing.h

```diff
@@ -0,0 +1,47 @@
+/*
+   +----------------------------------------------------------------------+
+   | Xdebug                                                               |
+   +----------------------------------------------------------------------+
+   | Copyright (c) 2002-2020 Derick Rethans                               |
+   +----------------------------------------------------------------------+
+   | This source file is subject to version 1.01 of the Xdebug license,   |
+   | that is bundled with this package in the file LICENSE, and is        |
+   | available at through the world-wide-web at                           |
+   | https://xdebug.org/license.php                                       |
+   | If you did not receive a copy of the Xdebug license and are unable   |
+   | to obtain it through the world-wide-web, please send a note to       |
+   | derick@xdebug.org so we can mail you a copy immediately.             |
+   +----------------------------------------------------------------------+
+   | Authors: Derick Rethans <derick@xdebug.org>                          |
+   |          Michael Voříšek <mvorisek@mvorisek.cz>                      |
+   +----------------------------------------------------------------------+
+ */
+
+#ifndef __XDEBUG_TIMING_H__
+#define __XDEBUG_TIMING_H__
+
+#define NANOS_IN_SEC 1000000000
+#define NANOS_IN_MICROSEC 1000
+
+#if PHP_WIN32
+	typedef void (WINAPI *WIN_PRECISE_TIME_FUNC)(LPFILETIME);
+#endif
+
+typedef struct _xdebug_nanotime_init {
+	uint64_t start_abs;
+	uint64_t start_rel;
+	uint64_t last_abs;
+	uint64_t last_rel;
+	#if PHP_WIN32
+		WIN_PRECISE_TIME_FUNC win_precise_time_func;
+		uint64_t win_freq;
+	#endif
+} xdebug_nanotime_init;
```

Comment:
I think I would have picked the name `xdebug_nanotime_context` instead. The struct isn't for initialising anything, but to store contextual information.

The `#if PHP_WIN32` section (and typedef above) also should not be indented as pre-process defines aren't control structures.

Reply:
Yes, `xdebug_nanotime_context` sounds also better to me...